packages/eslint-plugin/src/rules/no-for-in-array.ts

```diff
@@ -45,3 +45,20 @@ export default createRule({
     };
   },
 });
+
+function isTypeArrayTypeOrUnionOfArrayTypes(
```

Comment:
(style) avoid having the identical name function but with different behavior compared to the `isTypeArrayTypeOrUnionOfArrayTypes` in utils?